docs/jetting.md

```diff
@@ -0,0 +1,709 @@
+       to have a declaration for each of your u3we() functions.  Every u3we()  function looks the same, e.g.
+	       u3_noun u3we_xxx(u3_noun);
+
+* edit the header file
+          ~/tlon/urbit/include/jets/q.h
+
+   to have a declaration for your u3qe() function.  u3qe() functions
+   can differ from each other, taking distinct numbers of u3_nouns and/or u3_atoms, e.g
+
+          u3_noun u3qe_yyy(u3_atom, u3_atom);
+          u3_noun u3qe_zzz(u3_noun, u3_noun, u3_atom, u3_atom);
+
+* create a new .c file to hold your jets (both the u3we_() and u3qe_() functions go in the same file).  For example, ~/tlon/urbit/jets/e/secp.c
+  (The choice of subdirectory under 'jets' (a through g) is XXX )
+   The new file should have three things:
+   1.   \#include "all.h"
+   1.   the new u3we() function
+   1.   the new u3qe() function
+
+* edit ~/tlon/urbit/jets/tree.c to register jet
+
+In the hoon code we hinted some leaf node functions (`ccc` in our example) and then hinted each parent node up to the root `aaa`).
+
+We need to replicate this structure in C.  Here's example C code to jet our above example Hoon:
+
+```
+			// 1: register a C func u3we_ccc()
+			static u3j_harm _143_hex_hobo_reco_d[] =
+			  {
+				{".2", u3we_ccc, c3y},
+				{}
+			  };  
+
+			// 2: that implements a jet for hoon arm 'ccc'
+			static u3j_core _143_hex_hobo_bbb_d[] =
+			  {
+				{ "ccc",            _143_hex_hobo_ccc_d },                         
+				{}
+			  };
+
+			// 3: ... that is inside a hoon arm 'bbb'
+			static u3j_core _143_hex_hobo_hobo_d[] =
+			  {
+				{ "bbb", 0,   _143_hex_hobo_bbb_d },                               
+				{}
+			  };
+
+			// 4: ... that is inside a hoon arm 'aaa'
+			static u3j_core _143_hex_d[] =
+			{ { "down", 0, _143_hex_down_d }, 
+			  { "lore", _143_hex_lore_a },
+			  { "loss", _143_hex_loss_a },
+			  { "lune", _143_hex_lune_a },
+			  { "coed", 0, _143_hex_coed_d },
+			  { "aes", 0, _143_hex_aes_d },
+			  { "hmac", 0, _143_hex_hmac_d },
+			  { "aaa", 0, _143_hex_hobo_d },                                       
+			  {}
+			};
+
+```
+
+There are 4 things steps here.  Let's look at each in turn.
+
+Section 1 names the C function that we want to invoke: 'u3we_ccc()'.
+The precise manner in which it does this is by putting entries in an
+array of 'u3j_harm's.  The first one specifies the jet; the second one
+is empty and serves as a null termination to the array.  The jet
+registration supplies two fields {".2", u3we_secp}, but this does not
+initialize all of the fields of u3j_harm.  Other fields can be
+specified.
+
+The first field, with value ".2" in this example, is "arm 2".  Why? XXX
+
+The second field, with value u3we_secp in this example, is a function
+pointer (to the C implementation of the jet).
+
+The third field (absent here) if a flag to turn on verification of C
+jet vs hoon at run time.  It can take value c3n (yes, verify) or c3y
+(no, don't verify).  If not present, it is set to don't verify.
```

Comment:
no, I don't think so.  I know what c3n and c3y mean. The actual meaning of the third field is "perfect / don't test", so the meanings of the loobeans are inverted. I've clarified:

```
The third field (absent here) is a flag to turn on verification of C
jet vs hoon at run time.  It can take value c3n (which means verify at run time) or c3y
(which means don't verify).  If not present, it is set to don't verify.

[N.B. The use of the flags seems inverted. Why is c3y ("yes") used to
turn OFF verification?  A: because the flag is actually "is the jet
already known to be perfect?" ]```

Reply:
Thanks, this is very clear -- I had forgotten the semantics of that flag. The c3n (yes) and c3y (no) format was making me twitch ;)